config/m4/compiler.m4

```diff
@@ -411,6 +411,31 @@ ADD_COMPILER_FLAG_IF_SUPPORTED([-funwind-tables],
                                [AS_MESSAGE([compiling without unwind tables])])
 
 
+#
+# ASAN support
+#
+AC_ARG_ENABLE([asan],
+        AS_HELP_STRING([--enable-asan], [Enable address sanitized check]),
+        [],
+        [enable_asan=no])
+
+AS_IF([test "x$enable_asan" = xyes],
+      [ADD_COMPILER_FLAG_IF_SUPPORTED([-fsanitize=address -fno-omit-frame-pointer],
+                                     [-fsanitize=address -fno-omit-frame-pointer],
+                                     [AC_LANG_SOURCE([[int main(int argc, char** argv){return 0;}]])],
+                                     [AS_MESSAGE([compiling with sanitizer])
+                                      BASE_CXXFLAGS="-fsanitize=address -fno-omit-frame-pointer $BASE_CXXFLAGS"
+                                      LDFLAGS="-fsanitize=address -fno-omit-frame-pointer $LDFLAGS"],
```

Comment:
do we really need it also for link flags?

Reply:
Yes we do.

See here: https://github.com/google/sanitizers/wiki/AddressSanitizer#using-addresssanitizer
> In order to use [AddressSanitizer](https://github.com/google/sanitizers/wiki/AddressSanitizer) you will need to compile and link your program using clang with the -fsanitize=address switch.